### ISSUE TITLE
Rotated table/query functionality?

### ISSUE THREAD

**Issue by thomasnield** [sqlitebrowser/sqlitebrowser#1906]
Describe the new feature
--------------------------

Can there be a functionality to transpose the results of a SQL query, so that columns can become rows and rows become columns? This is helpful for viewing very wide tables with many fields. 

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.
No.

#### Does this feature exist in another product or project? Please provide a link.

SQL clients like SQuirrL and DataGrip support this feature: 
https://www.jetbrains.com/help/datagrip/viewing-query-results.html#transposing_table

#### Do you have a screenshot? Please add screenshots to help explain your idea.

![](blob:https://www.jetbrains.com/d6dbb01a-fd19-454a-ba4b-2eeca2f0fc34)

**Comment by justinclift** [comment]
Interesting idea.  It'd probably need to have some safety checks, but the concept itself seems ok. :smile:

**Comment by thomasnield** [comment]
Yeah, I doubt it'd do anything disruptive to your codebase. It's useful for analytical tasks with extremely wide tables.

**Comment by scottfurry** [comment]
Just an off-the-cuff observation, but there would probably be some kind of temporary table involved. The next question is what to do with the "transformed data" when the user is done. Save it? Save the SQL that does the transformation? Or save the temporary table? And down the "rabbit hole" we descend...

A couple of, what appears to be, useful background reading links:
https://stackoverflow.com/questions/1237068/pivot-in-sqlite
https://modern-sql.com/use-case/pivot

**Comment by mgrojo** [comment]
The proposed feature is a presentation change, there is no, or there shouldn't be, any transformed data or SQL query involved. The only change is between the data obtained in the query and the Qt model used to display the table view, or better between the Qt table model and the Qt table view.

Here there is a solution for transposing a Qt table view by using a proxy model: https://stackoverflow.com/questions/21653253/how-to-change-orientation-of-qt-tableview

**Comment by MKleusberg** [comment]
There is even a proxy model provided by Qt for this purpose: https://doc.qt.io/qt-5/qtransposeproxymodel.html

There are going to be a few problems regarding at least inserting, deleting, duplicating, and filtering rows though.
